app/models/post.rb

```diff
@@ -9,10 +9,8 @@ class Post < ApplicationRecord
   has_many :likes, dependent: :destroy
 
   def self.current_user_and_friends_posts(user)
-    frnds = FriendRequest.where(user_id: user.id, confirmed: true).pluck(:friend_id)
-    inverse = user.friends.pluck(:user_id)
-    frnds_and_inverse = frnds + inverse
-    frnds_and_inverse << user.id
-    Post.where(user_id: frnds_and_inverse).ordered_by_most_recent.includes(:user)
+    frnds = user.friends.pluck(:user_id)
+    frnds << user.id
+    Post.where(user_id: frnds).ordered_by_most_recent.includes(:user)
```

Comment:
From the `User` model, you could write this method,
```ruby
def friends_and_own_posts
  Post.where(user: (self.friends + self))
  # This will produce SQL query with IN. Something like: select * from posts where user_id IN (1,45,874,43);
end
```
and simply call it.

Reply:
Your suggested way didn't work as we got an error saying that `self.friends + self` can't be converted to an array. I think `where` is expecting an array of users to be able to return posts that belongs to users in the array.
Our code works fine as it supplies an array containing the id of the users whose posts we want to display.